server_src/utils/npm.utils.js

```diff
@@ -1,37 +1,39 @@
 const util = require("util");
 const exec = util.promisify(require("child_process").exec);
 
-async function doWeHaveMissingPackages() {
-  const { stdout, stderr } = await exec("npm outdated --production");
+async function returnListOfMissingPackages() {
+  try {
+    const outdatedPackageJsonList = await exec(
+      "npm outdated --production --json"
+    );
+    const outdatedPackageListParsed = JSON.parse(
+      outdatedPackageJsonList.stdout
+    );
+    if (outdatedPackageListParsed) {
+      const keys = Object.keys(outdatedPackageListParsed);
+      const missingPackageList = [];
 
-  if (stderr) {
-    throw `Could not reliably check for outdated npm packages, is npm installed? | ${stderr}`;
-  }
-
-  if (stdout) {
-    const matchMissingPackages = new RegExp("((.*?)MISSING.*)", "g");
-    const missingPackages = stdout.match(matchMissingPackages);
-    if (!!missingPackages) {
-      return missingPackages.length > 0;
-    } else {
-      return false;
+      keys.forEach((key, index) => {
+        if (!outdatedPackageListParsed[key].current) {
+          missingPackageList.push(key);
+        }
+      });
+      return missingPackageList;
```

Comment:
After grabbing the missing list keys and cycling through them I check for the "current" key missing on the returned list and push the key which is the name of the package to the array.

Reply:
Object.entries() not better readable/understandable?

Reply:
I didn't want a key value pair, only the key really.